test/cpp/end2end/test_service_impl.cc

```diff
@@ -464,44 +526,247 @@ Status TestServiceImpl::BidiStream(
   return Status::OK;
 }
 
-namespace {
-int GetIntValueFromMetadataHelper(
-    const char* key,
-    const std::multimap<grpc::string_ref, grpc::string_ref>& metadata,
-    int default_value) {
-  if (metadata.find(key) != metadata.end()) {
-    std::istringstream iss(ToString(metadata.find(key)->second));
-    iss >> default_value;
-    gpr_log(GPR_INFO, "%s : %d", key, default_value);
-  }
+void CallbackTestServiceImpl::RequestStream(
+    ServerContext* context, EchoResponse* response,
+    experimental::ServerCallbackReader<EchoRequest>* reader) {
+  class Reactor : public ::grpc::experimental::ServerReadReactor {
+   public:
+    Reactor(ServerContext* context,
+            experimental::ServerCallbackReader<EchoRequest>* reader,
+            EchoResponse* response)
+        : ctx_(context), reader_(reader), response_(response) {
+      reader->BindReactor(this);
```

Comment:
Is this always the intended place to bind? Any way this could be done in the base class ctor?